src/GovUk.Education.ExploreEducationStatistics.Common/Services/PostgreSqlRepository.cs

```diff
@@ -14,6 +14,38 @@ namespace GovUk.Education.ExploreEducationStatistics.Common.Services;
 
 public class PostgreSqlRepository : IPostgreSqlRepository
 {
+#pragma warning disable EF1002
+    public async Task<bool> KeyExistsAtJsonbPath<TDbContext, TRowId>(TDbContext context,
+        JsonbPathRequest<TRowId> request,
+        string keyValue,
+        CancellationToken cancellationToken = default)
+        where TDbContext : DbContext
+    {
+        var jsonPathParam = new NpgsqlParameter("jsonPath", request.PathSegments);
+        var keyValueParam = new NpgsqlParameter("keyValue", keyValue);
+        var rowIdParam = new NpgsqlParameter("rowId", request.RowId);
+
+        // We use the COALESCE function below to ensure that we're testing the existence of
+        // the given key against a non-null JSON fragment. Without this, an exception is thrown
+        // whereas we are wanting to return false. 
+        var response = await context
+            .Set<JsonBool>()
+            .FromSqlRaw(
+                sql: $"""
+                      SELECT COALESCE("{request.JsonbColumnName}" #> @jsonPath, '[]'::jsonb)
```

Comment:
Yup, this means that if `{request.JsonbColumnName}" #> @jsonPath` resolves to NULL, then this will fall back to a non-null JSON object for the key to be compared against.

Without this fallback, it goes kaboom!

Reply:
but doesn't `COALESCE` return the first non-null value in the parameter list? And if none are non-null, then it returns null itself? So I thought it would have the same behaviour?